persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -295,6 +295,13 @@ public Authenticator getAuthenticator() {
             internalOptions.prepare(prepared.boundNames);
             CQLStatement statement = prepared.statement;
 
+            if (internalOptions.getValues().size() != statement.getBoundTerms()) {
+              throw new org.apache.cassandra.exceptions.InvalidRequestException(
+                  String.format(
+                      "there were %d markers(?) in CQL but %d bound variables",
```

Comment:
I unified the errors across all persistence backends. The original error is `throw new InvalidRequestException("Invalid amount of bind variables");` in `QueryProcessor` for simple statements for C* 3.11 and 4.0, but not for DSE. This seems more descriptive.